snuba/migrations/generation.py

```diff
@@ -0,0 +1,29 @@
+import datetime
+from dataclasses import dataclass
+
+from jinja2 import Environment, PackageLoader, select_autoescape
+
+from snuba.migrations.groups import MigrationGroup
+
+env = Environment(
+    loader=PackageLoader("snuba", package_path="migrations"),
+    autoescape=select_autoescape(),
+)
+
+template = env.get_template("migration_template.py.jinja")
+
+
+@dataclass(frozen=True)
+class GenerationResult:
+    filename: str
+    groups_entry: str
+
+
+def generate(group_enum: MigrationGroup, description: str) -> GenerationResult:
+    group = group_enum.value
+    transformed_description = description.lower().replace(" ", "_")
+    filesystem_friendly_timestamp = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
```

Comment:
Do we need the granularity upto seconds. Can't we do with just the date? The number just becomes too big in my opinion with that much granularity.

Reply:
Well the goal is that it should generate timestamps in order to create an increasing unique key. So I think one day isn't enough granularity (@untitaker and I created a migration in the same group on the same day recently) but maybe we could go down to the minute.

Also open to other ideas for this, I'm really just copying what Rails does for file naming in its migration system.

Reply:
Could you answer the following question to provide more clarity. 

When you do encounter a conflict on the migrations, are the steps like the following?
1. Rename the file using git.
2. Rename the migration in the migrations list.

Or is there more to it?

Reply:
yeah, that's exactly what you do (assuming 2 of the same version never make it to master, and then you can't rename)

Reply:
My main issue with switching over to the date is that we now have two conventions to contend with in our codebase, since we cannot go back and rename the older migrations to entirely use the new one. I feel that adds a lot of confusion. Fwiw Sentry/Django migrations uses the same naming convention we have here with the incrementing number prefix. An alternative idea to guard against this could be to add test or some other CI check that fails on any migration with a duplicate prefix in the same group.

Reply:
Migrations in Sentry, AFAIK, even go out of their way to force a merge-conflict by having this lockfile around. I wonder whether this was done because we couldn't switch to a different sequence format (like Rails) and had to ensure unique IDs, or because we actually wanted to make the author double-check whether migrations conflict.